assets/src/application/services/apollo/mutations/useMutationWithFeedback.ts

```diff
@@ -0,0 +1,87 @@
+import { useCallback, useRef } from 'react';
+import { MutationTuple, useMutation } from '@apollo/react-hooks';
+import { DocumentNode } from 'graphql';
+import { OperationVariables } from 'apollo-client';
+import { v4 as uuidv4 } from 'uuid';
+
+import { MutationType } from '@application/services/apollo/mutations';
+import { SystemNotificationsToaster } from '@application/services/toaster/types';
+
+interface MutationWithFeedbackArgs {
+	typeName: string; // e.g. "Datetime", "Ticket", "PriceType"
+	mutation: DocumentNode;
+	mutationType: MutationType;
+	toaster: SystemNotificationsToaster;
+}
+
+type MutationWithFeedback = <TData = any, TVariables = OperationVariables>(
+	args: MutationWithFeedbackArgs
+) => MutationTuple<TData, TVariables>['0'];
+
+const useMutationWithFeedback: MutationWithFeedback = ({ typeName, mutation, mutationType, toaster }) => {
+	// generate a toaster key that sustains re-renders
+	const toasterKey = useRef(uuidv4());
+
+	const key = toasterKey.current;
+
+	/**
+	 * Get the toaster message based upon typeName and mutationType
+	 */
+	const getToasterMessage = useCallback(
+		(suffix: 'ing' | 'ed' = 'ing') => {
+			// For example "CREATE" will become "creating" or "created"
+			const verb = mutationType.toLowerCase().replace(/e$/, suffix);
```

Comment:
interesting :+1: 
almost forgot that you are also a regex master

Reply:
Regex was the first thing that I learnt in programming to manipulate some text files 😄